mutters-ink-bot/src/main/java/com/rabidgremlin/mutters/bot/ink/SessionUtils.java

```diff
@@ -76,16 +82,28 @@ public static void saveInkStoryState(Session session, StoryState storyState)
    * Gets the user's current Ink story state from the session.
    * 
    * @param session The session.
-   * @param storyState The story state.
+   * @param story The story.
+   * @param storyJson The JSON string for the story. Used to temporary story version check hack. See TestSessionRestore.
    */
-  public static void loadInkStoryState(Session session, StoryState storyState)
+  public static void loadInkStoryState(Session session, Story story, String storyJson)
   {
+    StoryState storyState = story.getState();  
+	  
     try
     {
+      Long storyJsonLength = (Long) session
+    	          .getAttribute(SLOT_PREFIX + "0987654321STORYJSONLENGTH1234567890");
       String stateJson = (String) session
-          .getAttribute(SLOT_PREFIX + "0987654321STORYSTATE1234567890");
+              .getAttribute(SLOT_PREFIX + "0987654321STORYSTATE1234567890");            
+                  
       if (stateJson != null)
       {
+    	// have state to restore so check that we have a story length and then check for size mismatch  
+        if (storyJsonLength == null || storyJsonLength != storyJson.length())
```

Comment:
I wonder if it might become an issue when you compare **Long** with **int**

Reply:
"The Internet" says it should be ok.... But since length() returns an int, I might change to using an Integer :)